src/FluentMigrator.Runner.Postgres/Generators/Postgres10/Postgres10Column.cs

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright (c) 2019, Fluent Migrator Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#endregion
+
+using FluentMigrator.Infrastructure.Extensions;
+using FluentMigrator.Model;
+using FluentMigrator.Postgres;
+using FluentMigrator.Runner.Generators.Postgres;
+
+using JetBrains.Annotations;
+
+using System;
+
+namespace FluentMigrator.Runner.Generators.Postgres10
+{
+    internal class Postgres10Column : PostgresColumn
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Postgres10Column"/> class.
+        /// </summary>
+        /// <param name="quoter">The Postgres quoter.</param>
+        /// <param name="typeMap">The Postgres type map.</param>
+        public Postgres10Column([NotNull] PostgresQuoter quoter, ITypeMap typeMap)
+            : base(quoter, typeMap)
+        {
+        }
+
+        private static string GetIdentityString(ColumnDefinition column)
+        {
+            string generationType;
+
+            switch (column.GetAdditionalFeature(PostgresExtensions.IdentityGeneration, PostgresGenerationType.ByDefault))
+            {
+                case PostgresGenerationType.Always:
+                    generationType = "ALWAYS";
+                    break;
+                case PostgresGenerationType.ByDefault:
+                    generationType = "BY DEFAULT";
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            return $"GENERATED {generationType} AS IDENTITY";
+        }
+
+        /// <inheritdoc />
+        protected override string FormatIdentity(ColumnDefinition column)
+        {
+            return column.IsIdentity ? GetIdentityString(column) : string.Empty;
+        }
+
+        public override string GenerateAlterClauses(ColumnDefinition column)
+        {
+            var clauses = new System.Collections.Generic.List<string>();
+            foreach (var action in AlterClauseOrder)
+            {
+                string columnClause = action(column);
+                if (!string.IsNullOrEmpty(columnClause))
+                    clauses.Add(string.Format("ALTER COLUMN {0} {1}", Quoter.QuoteColumnName(column.Name), columnClause));
+            }
+
+            return string.Join(", ", clauses.ToArray());
+        }
+
+        /// <inheritdoc />
+        protected override string FormatType(ColumnDefinition column)
+        {
+            if (!column.Type.HasValue)
```

Comment:
For anyone else looking at this PR, this method needs to be overridden since PostgresColumn implements Identity columns by changing the type to `serial` or `bigserial` something that obviously isn't appropriate for a Postgres 10 Identity column definition.  This override here duplicates the definition of `ColumnBase.FormatType` which is skipping the base `PostgresColumn.FormatType` and going up to it's base class definition.

Rather than that duplication, I simply exposed it in PostgresColumn like so:
```
        protected string ColumnBaseFormatType(ColumnDefinition column)
        {
            return base.FormatType(column);
        }
```
So that I could call `base.ColumnBaseformatType` in this class and avoid the code duplication.  I couldn't think of a decent name for it, but I'm a big fan of avoiding code duplication, particularly from an ongoing maintenance point of view.  It's too easy to make a change or bugfix in one place and not be aware of all the other duplicated copies of that code that ought to be updated similarly, so if we can avoid that whenever possible, I think it's good practice to do so.